exchanges/order/limits.go

```diff
@@ -0,0 +1,365 @@
+	marketMaxQty            float64
+	marketStepIncrementSize float64
+	maxTotalOrders          int64
+	maxAlgoOrders           int64
+	sync.RWMutex
+}
+
+// Conforms checks outbound parameters
+func (t *Limits) Conforms(price, amount float64, orderType Type) error {
+	if t == nil {
+		// For when we return a nil pointer we can assume there's nothing to
+		// check
+		return nil
+	}
+
+	t.RLock()
+	defer t.RUnlock()
+	if t.minPrice != 0 && price < t.minPrice {
+		return fmt.Errorf("%w min: %f suppplied %f",
+			ErrPriceExceedsMin,
+			t.minPrice,
+			price)
+	}
+	if t.maxPrice != 0 && price > t.maxPrice {
+		return fmt.Errorf("%w max: %f suppplied %f",
+			ErrPriceExceedsMax,
+			t.maxPrice,
+			price)
+	}
+
+	if t.stepIncrementSizePrice != 0 {
+		increase := 1 / t.stepIncrementSizePrice
+		if math.Mod(price*increase, t.stepIncrementSizePrice*increase) != 0 {
+			return fmt.Errorf("%w stepSize: %f suppplied %f",
+				ErrPriceExceedsStep,
+				t.stepIncrementSizePrice,
+				price)
+		}
+	}
+
+	if t.minAmount != 0 && amount < t.minAmount {
+		return fmt.Errorf("%w min: %f suppplied %f",
+			ErrAmountExceedsMin,
+			t.minAmount,
+			price)
+	}
+
+	if t.maxAmount != 0 && amount > t.maxAmount {
+		return fmt.Errorf("%w min: %f suppplied %f",
+			ErrAmountExceedsMax,
+			t.maxAmount,
+			price)
+	}
+
+	if t.stepIncrementSizeAmount != 0 {
+		increase := 1 / t.stepIncrementSizeAmount
+		if math.Mod(amount*increase, t.stepIncrementSizeAmount*increase) != 0 {
+			return fmt.Errorf("%w stepSize: %f suppplied %f",
+				ErrAmountExceedsStep,
+				t.stepIncrementSizeAmount,
+				amount)
+		}
+	}
+
+	if t.minNotional != 0 && (amount*price) < t.minNotional {
+		return fmt.Errorf("%w minimum notional: %f value of order %f",
+			ErrNotionalValue,
+			t.minNotional,
+			amount*price)
+	}
+
+	// Multiplier checking not done due to the fact we need coherence with the
+	// last average price (TODO)
+	// t.multiplierUp will be used to determine how far our price can go up
+	// t.multiplierDown will be used to determine how far our price can go down
+	// t.averagePriceMinutes will be used to determine mean over this period
+
+	// Max iceberg parts checking not done as we do not have that
+	// functionality yet (TODO)
+	// t.maxIcebergParts // How many components in an iceberg order
+
+	if orderType == Market {
+		if t.marketMinQty != 0 &&
+			t.minAmount < t.marketMinQty &&
+			amount < t.marketMinQty {
+			return fmt.Errorf("%w min: %f suppplied %f",
+				ErrMarketAmountExceedsMin,
+				t.marketMinQty,
+				amount)
+		}
+		if t.marketMaxQty != 0 &&
+			t.maxAmount > t.marketMaxQty &&
+			amount > t.marketMaxQty {
+			return fmt.Errorf("%w max: %f suppplied %f",
+				ErrMarketAmountExceedsMax,
+				t.marketMaxQty,
+				amount)
+		}
+		if t.marketStepIncrementSize != 0 && t.stepIncrementSizeAmount != t.marketStepIncrementSize {
+			increase := 1 / t.marketStepIncrementSize
+			if math.Mod(amount*increase, t.marketStepIncrementSize*increase) != 0 {
```

Comment:
We can remove the creation of increase and just divide by marketStepIncrementSize when needed. Here and elsewhere please

Reply:
That's done, if I do it in the conform function the rounding issues get worse, can you check that out when you have time?

Reply:
Seems like the way you are doing it fixes it 🤷‍♂️ the float rounding issues result in 1.00099999999999988987 1.00100000000000011191 which don't end up matching. But the approach you take where you create increase works and the result matches the value 1.1 as a float holds